tgui/packages/tgui/interfaces/Secrets.js

```diff
@@ -0,0 +1,723 @@
+            fluid
+            content="Everyone is the traitor"
+            onClick={() => act("traitor_all")} />
+        </NoticeBox>
+      </Flex.Item>
+      <Flex.Item>
+        <NoticeBox danger>
+          <Button
+            color="red"
+            align="center"
+            icon="brain"
+            fluid
+            content="Make all players brain damaged"
+            onClick={() => act("massbraindamage")} />
+        </NoticeBox>
+      </Flex.Item>
+      <Flex.Item>
+        <NoticeBox danger>
+          <Button
+            color="black"
+            align="center"
+            icon="fire"
+            fluid
+            content="The floor is lava! (DANGEROUS: extremely lame)"
+            onClick={() => act("floorlava")} />
+        </NoticeBox>
+      </Flex.Item>
+      <Flex.Item>
+        <NoticeBox danger>
+          <Button
+            color="black"
+            align="center"
+            icon="tired"
+            fluid
+            content="Chinese Cartoons! (DANGEROUS: no going back, also fuck you)"
+            onClick={() => act("anime")} />
+        </NoticeBox>
+      </Flex.Item>
+      <Flex.Item>
+        <Flex>
+          <Flex.Item width="240px" mr={0.25}>
+            <NoticeBox danger>
+              <Button
+                color="red"
+                align="center"
+                icon="cat"
+                fluid
+                content="Mass Purrbation"
+                onClick={() => act("masspurrbation")} />
+            </NoticeBox>
+          </Flex.Item>
+          <Flex.Item grow={1} ml={0.25}>
+            <NoticeBox info>
+              <Button
+                color="blue"
+                align="center"
+                icon="user"
+                fluid
+                content="Mass Remove Purrbation"
+                onClick={() => act("massremovepurrbation")} />
+            </NoticeBox>
+          </Flex.Item>
+        </Flex>
+      </Flex.Item>
+      <Flex.Item>
+        <Flex justify="space-between">
+          <Flex.Item width="240px" mr={0.25}>
+            <NoticeBox danger>
+              <Button
+                color="red"
+                align="center"
+                icon="flushed"
+                fluid
+                content="Fully Immerse Everyone"
+                onClick={() => act("massimmerse")} />
+            </NoticeBox>
+          </Flex.Item>
+          <Flex.Item grow={1} ml={0.25}>
+            <NoticeBox info>
+              <Button
+                color="blue"
+                align="center"
+                icon="sync-alt"
+                fluid
+                content="Shatter the Immersion"
+                onClick={() => act("unmassimmerse")} />
+            </NoticeBox>
+          </Flex.Item>
+        </Flex>
+      </Flex.Item>
+    </Flex>
+  );
+};
+
+export const Secrets = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    is_debugger,
+    is_funmin,
+  } = data;
+  const TAB2NAME = [
```

Comment:
Extract it to above Secrets function, to not reconstruct this array every time Secrets re-render.